dev/com.ibm.ws.microprofile.faulttolerance.2.0/test/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/FallbackStateImplTest.java

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.faulttolerance20.state.impl;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import com.ibm.ws.microprofile.faulttolerance.impl.policy.FallbackPolicyImpl;
+import com.ibm.ws.microprofile.faulttolerance.spi.MetricRecorder;
+import com.ibm.ws.microprofile.faulttolerance.utils.DummyMetricRecorder;
+import com.ibm.ws.microprofile.faulttolerance20.impl.MethodResult;
+
+@SuppressWarnings("restriction") // Unit test accesses non-exported *PolicyImpl classes
+public class FallbackStateImplTest {
+
+    private static MetricRecorder dummyMetrics = DummyMetricRecorder.get();
+
+    @Test
+    public <R> void testFallback() throws InterruptedException, InstantiationException, IllegalAccessException {
+        FallbackPolicyImpl policy = new FallbackPolicyImpl();
+        FallbackStateImpl state = new FallbackStateImpl(policy, dummyMetrics);
+
+        MethodResult<R> resultWithException = MethodResult.failure(RuntimeException.class.newInstance());
+        assertTrue("Fallback should be applied if an exception is thrown, according to Fallback's default applyOn() and skipOn() values",
+                   state.shouldApplyFallback(resultWithException));
+
+        MethodResult<R> resultNoException = MethodResult.success(null);
+        assertFalse("Fallback should NOT be applied if an exception is NOT thrown, according to Fallback's default applyOn() and skipOn() values",
+                    state.shouldApplyFallback(resultNoException));
```

Comment:
Could make these assertion messages shorter. Just "Fallback should (NOT) be applied if an exception is (NOT) thrown" would cover what the test is checking.